client/components/data/query-site-keyrings/index.js

```diff
@@ -0,0 +1,56 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingSiteKeyrings } from 'state/site-keyrings/selectors';
+import { requestSiteKeyrings } from 'state/site-keyrings/actions';
+
+class QuerySiteKeyrings extends Component {
+	componentWillMount() {
+		this.requestKeyrings( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
```

Comment:
We should use `getDerivedStateFromProps()` instead as this lifecycle method is going to be deprecated soon.